Maye mech branch

Mayegelt opened the pull request:
Created SE's for all units, with unit profiles and normal constraints and pts values and options

Only have these left to finish from units:
Krios Battle (only requires Rapid Sub-type category)
Krios Venator (only requires Rapid Sub-type category)
Brass Scorpion (only requires Unstoppable Sub-type category)
Kytan (only requires Unstoppable Sub-type category)

Got SE's and Rules for all High Arcanas
Apex Detachments (Need additional PRIME slots in GST & needs special rules enforcing)

Setup Roots for all units and Restricted Decimator, Blood Slaughterer, Brass Scorpion, Kytan, Archmagos Scoria, Archmagos Kraykavac to Traitor only

Mayegelt commented:
Without some additions to the main files the Mech is as ready as I think I can get it.
I edited my post on https://github.com/BSData/horus-heresy-3rd-edition/issues/4 with a basic list of what is still needing to be done.
The main bits are needing a few things like a Cybertheurgy Profile. Having a Prime category for all slots. Linking up all the Prime stuff for any unit which can have it.
Working out how the Detachment thing is working over multiple armies (but I think that is a larger discussion for the future).
Adding additional category like "Vehicle" and "Rapid" to the GST so they can be put on units. This is key to one of the Prime Benefits in Mech.

The4D6 commented on Mechanicum.cat:
As per 1.2 of the Implementation Guide, please leave a top level comment anywhere you have variable special rules to indicate what they are at a glance

See pic for example:
<img width="367" height="116" alt="image" src="https://github.com/user-attachments/assets/281c4b28-23a0-41ae-9eba-64a65fba3b5d" />

Mayegelt replied:
Fixed.

The4D6 commented on Mechanicum.cat:
What is theurgika maximas and why is it a selection available to the archmagos?

Mayegelt replied:
Opps pasted in the wrong place first...
This is for the Archemandrite Archmagos and Archmagos on Abeyant as an Arcana Option. Uniquely it is the only one which costs pts. The other ones are Gambits and Prime selections. All are restricted to only certain units.
Additionally Scoria and Draykavac also get it.

The4D6 commented on Mechanicum.cat:
Please flatten the Power Weapon selection under Mechanicum Weapons

Mayegelt replied:
This is an option for Archimendrite Archmagos and Archmagos on Abeyant only. It is their "Arcana Option" Similar to others get some Prime Benefits or Gambits as an option. But they have to pay for there one.

Mayegelt replied:
Done. I had thought it did it automatically

The4D6 commented on Mechanicum.cat:
The mechanicum wargear list should be ticky boxes, not rollers

The4D6 commented on Mechanicum.cat:
Triaros does not need to have two separate instances of centreline volkite caliver
Please only include the profile once, we do not want to surface this entry box
<img width="372" height="71" alt="image" src="https://github.com/user-attachments/assets/75866202-4bb1-4ec6-aad3-c5278e3ed3ea" />

Same on Karacnos lightnign locks
<img width="397" height="74" alt="image" src="https://github.com/user-attachments/assets/061eb962-4d3a-47a7-a188-65218f27b19e" />

Mayegelt replied:
Fixed. Now just has 1 under the title of it saying we have "2 X mounted...."

The4D6 commented on Mechanicum.cat:
Thallax have no upper limit on special weapon selection per model, it should be ticky boxes not a roller.

Praetorian can be selected on every model

Mayegelt replied:
Fixed weapon. The Praetorian is an upgrade to 1 model. It has a "Max 1 selection in Thallax Cohort" restriction

The4D6 commented on Mechanicum.cat:
Krios tanks have HKMs selected by default

Mayegelt commented:
> RE: Apex detachments, I'd suggest adding them in tot he GST anyway, and as we've said, there will be no detachment validation in the initial launch
> 
> However I'd rather have the detachments in for now until we work it all out later

Moved them into the Crusade and Allied detachments.
Added the Primes for Support, Heavy Assault, Recon, Fast attack... to fill out all the slots as required by Mech and other RoW. Didn't do one for High Command, LoW or Warlord, as they are likely to not get one.
Added the Rapid Sub-Type Category.
Added the Vehicle Model Type Category as well as one of the prime benefits for Mech will require it.

I haven't fully coded getting the prime stuff on Mech yet. That is a future thing.
Got the Cyber Profile now as well as Gambit. But haven't written the Cyber stuff out yet.

The4D6 reviewed (changes requested):
You have broken the faction filter on selecting detachments somehow
<img width="858" height="836" alt="image" src="https://github.com/user-attachments/assets/c10f9938-8b14-4db0-ac1a-65e3ce4575af" />
The dropdown should be locked to the primary catalogue but now it is not

I cannot immediately see how this has happened, but it needs fixing

# Edit: Fixed it
